class.jetpack-client.php

```diff
@@ -285,4 +300,36 @@ static function wpcom_json_api_request_as_blog( $path, $version = self::WPCOM_JS
 		return Jetpack_Client::remote_request( $validated_args, $body );
 	}
 
+	/**
+	 * Takes an array or similar structure and recursively turns all values into strings. This is used to
+	 * make sure that body hashes are made ith the string version, which is what will be seen after a
+	 * server pulls up the data in the $_POST array.
+	 *
+	 * @param array|mixed $data
+	 *
+	 * @return array|string
+	 */
+	public static function _stringify_data( $data ) {
```

Comment:
Yes, that is exactly what we don't want. Remember, we're not ever unserializing this data. We're turning it into a hash and then using that hash to verify that the data on the other side wasn't tampered with. Unfortunately transporting via HTTP automatically tampers with it by changing all simple types to arrays and strings, because that is all an http encoded string knows.